api/web_conn.go

```diff
@@ -57,14 +60,26 @@ func (c *WebConn) readPump() {
 		return nil
 	})
 
+	if c.SessionToken == "" {
+		go func() {
+			time.Sleep(AUTH_TIMEOUT)
+			if c.SessionToken == "" {
+				l4g.Debug("websocket.read: did not authenticate in time, ip=%v", c.WebSocket.RemoteAddr())
+				c.WebSocket.Close()
+			}
+		}()
+	} else {
+		HubRegister(c)
```

Comment:
This kind of makes me mervous.  Why not allow the normal hub registration and skip sending if it isn't set.

Then in readPump you can have a timeout that's similar to the ticker in writePump, which can do the proper normal cleanup flow.